app/MoonShine/Resources/SkillResource.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace App\MoonShine\Resources;
+
+use App\Models\Skill;
+use MoonShine\Fields\ID;
+use MoonShine\Fields\Text;
+use MoonShine\Fields\Select;
+use MoonShine\Resources\Resource;
+use MoonShine\Actions\FiltersAction;
+use Illuminate\Database\Eloquent\Model;
+
+class SkillResource extends Resource
+{
+    public static string $model = Skill::class;
+
+    public static string $title = 'Skills';
+
+    public function fields(): array
+    {
+        return [
+            ID::make()->sortable(),
+            Text::make('Name'),
+            Text::make('Icon'),
+            Select::make('Parent skill')->nullable()->options($this->GetSkillOptions())->searchable()
```

Comment:
Select::make('Parent skill')->nullable()->options($this->GetSkillOptions())->searchable()
Select::make('Parent skill')->nullable()->default(callback yozas);

Reply:
Callback hozir nimaga kerak tushunmadim, select ishlatilgani sababi tushunarliro bolishi uchun ierarxiya bilan korsatish uchun